internal/app/balance/repository.go

```diff
@@ -0,0 +1,160 @@
+package balance
+
+import (
+	"context"
+	"database/sql"
+	"errors"
+	"fmt"
+	"github.com/jackc/pgx/v5/pgconn"
+	"github.com/korol8484/gofermart/internal/app/domain"
+	"strings"
+	"time"
+)
+
+type Repository struct {
+	db *sql.DB
+}
+
+func NewBalanceRepository(db *sql.DB) *Repository {
+	return &Repository{db: db}
+}
+
+func (r *Repository) GetUserWithdrawals(ctx context.Context, userID domain.UserID) ([]*domain.Balance, error) {
+	rows, err := r.db.QueryContext(
+		ctx,
+		`SELECT * FROM balance b WHERE b.user_id = $1 AND b.type = $2 ORDER BY b.created_at DESC;`,
+		userID,
+		domain.BalanceTypeWithdrawn,
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	if rows.Err() != nil {
+		return nil, rows.Err()
+	}
+
+	defer rows.Close()
+
+	var win []*domain.Balance
```

Comment:
Мне кажется в общем случае нет необходимости использовать тип указателя, так как слайс и так передается по ссылке.
```suggestion
	var win []domain.Balance
```